source/defclass-star.lisp

```diff
@@ -603,18 +612,31 @@ Example:
                                  forms))
          ;; NOTE: `set-difference' can shuffle the order of elements, thus `remove-if'.
          (method-body (remove-if (lambda (f) (member f options :test #'equal)) forms))
+         (export-generic-name-p (first (member :export-generic-name-p options :key #'first)))
+         (options (remove export-generic-name-p options :test #'equal))
+         (export-generic-name-p (second export-generic-name-p))
```

Comment:
A bit convoluted and imperative way of setting options...

Reply:
Yeah, but I was unable to come up with a much better way unfortunately :(

Reply:
I've changed the logic a bit, should be better now.